logic/src/commonMain/kotlin/com/wire/kalium/logic/data/publicuser/UserSearchApiWrapper.kt

```diff
@@ -36,36 +50,67 @@ internal class UserSearchApiWrapperImpl(
         domain: String,
         maxResultSize: Int?,
         searchUsersOptions: SearchUsersOptions
-    ): Either<NetworkFailure, UserSearchResponse> {
-
-        val searchResponse = wrapApiRequest {
+    ): Either<NetworkFailure, UserSearchResponse> =
+        wrapApiRequest {
             userSearchApi.search(
                 UserSearchRequest(
                     searchQuery = searchQuery,
                     domain = domain,
                     maxResultSize = maxResultSize
                 )
             )
+        }.map { userSearchResponse ->
+            filter(userSearchResponse, searchUsersOptions)
         }
 
-        return if (searchUsersOptions.conversationExcluded is ConversationMemberExcludedOptions.ConversationExcluded) {
-            val conversationMembersId = conversationDAO.getAllMembers(
-                qualifiedID = idMapper.toDaoModel(qualifiedID = searchUsersOptions.conversationExcluded.conversationId)
-            ).firstOrNull()?.map { idMapper.fromDaoModel(it.user) }
+    private suspend fun filter(
```

Comment:
Hmm, if we want to give other apps flexibility to decide if they want to filter own user from search this functionality should be in use cases

Reply:
good point

Reply:
@vitorhugods what do you think? 😄

Reply:
Indeed.

I think it would be just way simpler to do it in the Search UseCase.

Inside the UseCase:
- get all results
- get self user (easily accessible)
- before returning all results, filter them out if wanted.

Instead, we're manipulating the search, and passing this logic all the way down to a `APIWrapper` unnecessarily, adding extra complexity and parameters to simple mechanisms.

We need to stop adding business logic to API, DAOs and Repositories.

Reply:
I honestly disagree with it, UserSearchApiWrapper responsibility is to extend the functionality of the API that does not support searching for the users that are not included in the conversation. If use cases or repositories need this functionality they simply can inject it, adding this logic to a use case would limit us to not being able to use it inside the repositories if needed in future, also whenever the back-end perhaps going to support searching for the users excluded from the conversation we simply can drop the filtering and just delegate the call. I do not see any benefit of moving all this filtering to a use case, but perhaps I oversee something ? Also filtering is not really a buisness logic to me ,it is more of a repository responsiblity to simply manipulate the data